docs: Add policymap pressure debugging guide (#27903)
cilium-team/cilium: Documentation/operations/troubleshooting.rst

your ``ingress`` and ``egress`` rules match what you expect? If not, the
reference to the errant rules can be found in the ``derived-from-rules`` node.

Policymap pressure and overflow
-------------------------------

> The most important step is finding out which node is experiencing policymap

Reviewer: “The most important step” for what?

pressure.

> The ``cilium_bpf_map_pressure{map_name="cilium_policy_*"}`` metric monitors the

Reviewer 2: ```suggestion
The ``cilium_bpf_map_pressure{map_name=~"cilium_policy_.*"}`` metric monitors the
```
Feel free to reject, but regex matching should be required for this wildcard to work.

Author: I've written these docs with https://github.com/cilium/cilium/pull/27367 in mind :)

Reviewer 2: Oh, slick.  :smile:  Carry on, then!

endpoint's BPF policymap pressure. This metric exposes the maximum BPF map
pressure on the node, meaning the policymap experiencing the most pressure on a
particular node.

Once the node is known, the troubleshooting steps are as follows:

1. Find the Cilium pod on the node experiencing the problematic policymap
   pressure and obtain a shell via ``kubectl exec``.
2. Use ``cilium policy selectors`` to get an overview of which selectors are
   selecting many identities. The output of this command as of Cilium v1.15
   additionally displays the namespace and name of the policy resource of each
   selector.

> 3. The type of selector tells us what sort of policy rule could be having an
>    impact. There are 3 types of selectors are explained below, each with an
>    specific steps depending on the selector.

Reviewer: ```suggestion
3. The type of selector tells you what sort of policy rule could be having an
   impact. The three existing types of selectors are explained below, each with
   specific steps depending on the selector.
```

Specific steps for what?

> 4. Consider bumping the policymap size as a last resort.

Reviewer 3: >as a last resort

What would be the implications of increasing policy map size besides increasing the memory footprint?

Author: Added some context here.

                                                                                                                                                                                                                16860328
    &LabelSelector{MatchLabels:map[string]string{any.name: netperf,k8s.io.kubernetes.pod.namespace: default,},MatchExpressions:[]LabelSelectorRequirement{},}           default/tofqdn-dns-visibility   1
    &LabelSelector{MatchLabels:map[string]string{cidr.1.1.1.1/32: ,},MatchExpressions:[]LabelSelectorRequirement{},}                                                    default/tofqdn-dns-visibility   1       16860329
    &LabelSelector{MatchLabels:map[string]string{cidr.1.1.1.2/32: ,},MatchExpressions:[]LabelSelectorRequirement{},}                                                    default/tofqdn-dns-visibility   1       16860330
    &LabelSelector{MatchLabels:map[string]string{cidr.1.1.1.3/32: ,},MatchExpressions:[]LabelSelectorRequirement{},}                                                    default/tofqdn-dns-visibility   1       16860331

> From the output above, we see that all 3 selectors are in use. The significant
> action here is to determine which selector is selecting the most identities,
> because the policy containing that selectors is the likely cause for the
> policymap pressure.

Reviewer: ```suggestion
From the output above, we see that all three selectors are in use. The significant
action here is to determine which selector is selecting the most identities,
because the policy containing that selector is the likely cause for the
policymap pressure.
```
